lib/rest/Twilio.js

```diff
@@ -249,18 +252,39 @@ Twilio.prototype.request = function request(opts) {
     headers.Accept = 'application/json';
   }
 
-  var uri = opts.uri;
-  if (this.region) {
-    var parts = _.split(uri, '.');
-
-    if (parts.length > 1 && !_.isEqual(parts[1], this.region)) {
-      uri = _.join(_.concat([parts[0], this.region], _.slice(parts, 1)), '.');
+  var product;
+  var edge;
+  var region;
+  var subdomain;
+  var domain;
+  var uri = new url.URL(opts.uri);
+  var parsedHostname = uri.hostname.split('.');
+  if (parsedHostname.length === 5) { // https://product.edge.region.subdomain.domain
```

Comment:
Think it's simpler to first get everything before `.twilio.com` (if it exists), then split the prefix into 3 parts (`[product, edgeOrRegion, region] = prefix.split('.')`), and then reconstruct it based on what's defined.

Reply:
I'll try that.